exploration/0477-quoted-literals.md

```diff
@@ -0,0 +1,192 @@
+  > ```
+  >
+  > Note that quoted literals cannot contain placeholders, making interpolating data into them impossible.
+  >
+  > ```
+  > -- This is impossible in MessageFormat 2.0.
+  > {{+button title=|Goodbye, {$userName}!|}Sign out{-button}}
+  > ```
+
+- Selector function implementers may want to support exotic characters in variant keys to effectively create "mini-DSLs" for the matching logic:
+
+  > ```
+  > match {$count :myNumber}
+  > when |<10| {A handful.}
+  > when * {Lots.}
+  > ```
+
+- Message authors may want to protect untranslatable strings:
+
+  > ```
+  > {Visit {|http://www.example.com| @translate=false}.}
+  > ```
+  >
+  > See [design proposal 0002](https://github.com/unicode-org/message-format-wg/blob/main/exploration/0002-expression-attributes.md).
+
+- Message authors may want to decorate substrings as being written in a particular language, different from the message's language, for the purpose of accessibility, text-to-speech, and semantic correctness.
+
+  > ```
+  > {The official native name of the Republic of Poland is {|Rzeczpospolita Polska| @lang=pl}.}
+  > ```
+  >
+  > See [design proposal 0002](https://github.com/unicode-org/message-format-wg/blob/main/exploration/0002-expression-attributes.md).
+
+- Developers may want to embed messages with quoted literals in code written in another programming language which uses single or double quotes to delimit strings.
+
+  > ```js
+  > let message = new MessageFormat("en", "{A message with {|a literal|}.}");
+  > ```
+
+- Developers and localization engineers may want to embed messages with quoted literals in a container format, such as JSON.
+
+  > ```json
+  > {
+  >   "msg": "{A message with {|a literal|}.}"
+  > }
+  > ```
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+- **[r1; high priority]** Minimize the need to escape characters inside literals. In particular, choose a delimiter that isn't frequently used in translation content. Having to escape characters inside literals is inconvenient and error-prone when done by hand, and it also introduces the backslash into the message, `\`, which is the escape introducer. The backslash then needs to be escaped too, when the message is embedded in code or containers. (This is how some syntaxes produce the gnarly `\\\`.)
+- **[r2; high priority]** Minimize the need to escape characters when embedding messages in code or containers. In particular, choose a delimiter that isn't frequently used as a string delimiter in programming languages and container formats. However, note that many programming languages also provide alternative ways of delimiting strings, e.g. _raw strings_ or triple-quoted literals.
+- **[r3; medium priority]** Minimize the need to change the message in other ways than to escape some of its characters (e.g. rephrase content, use typographic apostrophes, or switch to using a second set of delimtiers).
```

Comment:
```suggestion
- **[r3; medium priority]** Minimize the incentive to avoid escaping by changing messages (e.g. rephrasing content, using typographic apostrophes, or switching outer delimiters).
```